repose-aggregator/artifacts/docker/build.gradle

```diff
@@ -46,28 +96,28 @@ docker {
     task("push${baseImage}ImageVersion", type: DockerPushImage) {
         dependsOn "tag${baseImage}ImageVersion"
         imageName = 'rackerlabs/repose'
-        tag = reposeVersion + tagSuffix
+        tag = currentVersion + tagSuffix
     }
 
     task("push${baseImage}ImageLatest", type: DockerPushImage) {
         dependsOn "tag${baseImage}ImageLatest"
+        onlyIf { isLatestDockerHubVersion(currentVersion) }
         imageName = 'rackerlabs/repose'
         tag = "latest${tagSuffix}"
     }
 
-    task("push${baseImage}Image") {
-        group = 'docker'
+    task("push${baseImage}Image", group: 'docker') {
         dependsOn "push${baseImage}ImageVersion", "push${baseImage}ImageLatest"
     }
 
     task("remove${baseImage}Image", type: DockerRemoveImage) {
         dependsOn "build${baseImage}Image"
-        mustRunAfter "tag${baseImage}ImageVersion", "tag${baseImage}ImageLatest", "push${baseImage}ImageVersion", "push${baseImage}ImageLatest"
+        mustRunAfter "tag${baseImage}ImageVersion", "tag${baseImage}ImageLatest", "push${baseImage}ImageVersion", "push${baseImage}ImageLatest", publish
         targetImageId { tasks["build${baseImage}ImageLocal"].getImageId() }
         force = true
     }
 
-    tasks["push${baseImage}ImageLatest"].onlyIf { isLatestDockerHubVersion(reposeVersion) }
+    publish.dependsOn "push${baseImage}Image"
```

Comment:
I'm not certain what this will do, this project generally doesn't interact with the publishing plugin (yes, i know technically it's defined).

Reply:
You're right, we already apply the publishing plugin to all sub-projects and wire every sub-project's `publish` task into the `release` task chain. Which means the `publish` task was already running on this sub-project as part of the `release` task chain.

I did question doing the wiring this way since we do not directly rely on the publishing plugin for this project, but since the task already runs and we have already wired the `publish` task correctly for all sub-projects, I figured this approach was more practical if not as relationally pure.

As an aside, part of the reason I ended up not using a source set to copy things was that I did not want to rely on the Java plugin which sets up the source set and source set directory I would have used.

Reply:
Thinking about it now, I do think I missed a step. I don't think I added any wiring to guarantee that the push image tasks run after `buildAll`. So while what I did added the task to the release task chain, it did not necessarily order it correctly.

In which case, I'll modify things a bit to not rely on the `publish` task and to instead insert the push image tasks between `buildAll` and `release`.

Reply:
`publish` itself depends on `buildAll`

Reply:
Sure, but prior to the last commit, I just had `publish` depending on `push${baseImage}Image`. So `publish` would depend on both `buildAll` and `push${baseImage}Image`, but `push${baseImage}Image` would not depend on `buildAll`. Thus, `push${baseImage}Image` had the potential to run before `buildAll`.